aks/locals.tf

```
@@ -1,6 +1,16 @@
 locals {
   use_managed_identity = try(var.metrics_storage.managed_identity_node_rg_name != null, false)
 
+  metrics_storage = var.metrics_storage != null ? {
+    storage_config = merge({ type = "AZURE" }, {
+      config = merge({
+        container       = var.metrics_storage.container
+        storage_account = var.metrics_storage.storage_account
+        },
+      local.use_managed_identity ? null : { storage_account_key = var.metrics_storage.storage_account_key })
```

Fixed indentation for readability (hopefully terraform fmt agrees) :
```suggestion
        local.use_managed_identity ? null : { storage_account_key = var.metrics_storage.storage_account_key }
      )
```

---

Terraform does not agree with you! :smile: